python/cuml/tests/experimental/test_filex.py

```diff
@@ -787,3 +787,49 @@ def test_predict_per_tree_with_vector_leaf(
         np.testing.assert_almost_equal(
             pred_per_tree, pred_per_tree_tl, decimal=3
         )
+
+
+@pytest.mark.parametrize("train_device", ("cpu", "gpu"))
+@pytest.mark.parametrize("infer_device", ("cpu", "gpu"))
+@pytest.mark.parametrize("n_classes", [2, 5, 25])
+@pytest.mark.skipif(not has_xgboost(), reason="need to install xgboost")
+def test_apply(train_device, infer_device, n_classes, tmp_path):
+    n_rows = 1000
+    n_columns = 30
+    num_boost_round = 10
+
+    with using_device_type(train_device):
+        X, y = simulate_data(
+            n_rows,
+            n_columns,
+            n_classes,
+            random_state=0,
+            classification=True,
+        )
+
+        model_path = os.path.join(tmp_path, "xgb_class.model")
```

Comment:
The `tmp_path` fixture should be a `pathlib.Path` object, so this should be equivalent:

```suggestion
        model_path = tmp_path / "xgb_class.model"
```